backend/src/main/java/de/christianporsch/backend/service/CardPileService.java

```
@@ -0,0 +1,69 @@
+        this.userRepository = userRepository;
+        this.magicCardInPileRepository = magicCardInPileRepository;
+        this.cardSearchService = cardSearchService;
+    }
+
+    public List<MagicCardInPile> findPileOfCardsByUser(String id) {
+        return userRepository.findUserById(id).getPileOfCards();
+    }
+
+    public Optional<MagicCardInPile> findMagicCardInPileById (String id){
+        return magicCardInPileRepository.findMagicCardInPileById(id);
+    }
+
+    public void addMagicCardToPile(MagicCardDto magicCardToAdd) {
+        Optional<MagicCard> magicCard = cardSearchService.findMagicCardById(magicCardToAdd.getId());
+
+        if (magicCard.isPresent()) {
+            User user = userRepository.findUserById("60d2f120c76f8707f38e9a99");
+            Optional<MagicCardInPile> magicCardInPile = user.getPileOfCards().stream().filter((card) -> magicCard.get().getId().equals(card.getId())).findFirst();
+            if(magicCardInPile.isPresent()){
+                magicCardInPile.get().setAmount(magicCardInPile.get().getAmount()+1);
+                magicCardInPileRepository.save(magicCardInPile.get());
+            }
+            else {
+                MagicCardInPile newMagicCardInPile = MagicCardInPile.builder()
+                        .id(magicCard.get().getId())
+                        .amount(1)
+                        .name(magicCard.get().getName())
+                        .oracle_text(magicCard.get().getOracle_text())
+                        .image_uris(magicCard.get().getImage_uris())
+                        .set_name(magicCard.get().getSet_name())
+                        .build();
+                user.getPileOfCards().add(newMagicCardInPile);
+                magicCardInPileRepository.save(newMagicCardInPile);
+            }
+            userRepository.save(user);
+        }
+         else {throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Error");}
```

I'd throw a more specific error ("NOT FOUND" / Magic card with id xxx not found)